src/transformers/trainer.py

```diff
@@ -2180,7 +2180,7 @@ def _load_best_model(self):
                     state_dict["_smp_is_partial"] = False
                     load_result = model.load_state_dict(state_dict, strict=True)
             else:
-                if _is_peft_model(model):
+                if _is_peft_model(unwrap_model(model)):
```

Comment:
weird that we never catched bugs here, in any case it is safe to put `unwrap_model(model)` here IMO

Reply:
Indeed - @AjayP13 can we add a test which passes with these fixes but fails on `main`?

Reply:
yes that would be great if possible!

Reply:
I unfortunately don't have consistent access to a multi-GPU machine (for which this test would require since FSDP requires >=2 GPUs), however, I did test this fix when I was running a job on a multi-GPU machine.

It seems PEFT + FSDP pathway is largely untested though, there is one FSDP test [here](https://github.com/huggingface/transformers/blob/main/tests/fsdp/test_fsdp.py). Would be great to get tests for this pathway in general, but probably won't be able to on my side, if someone working on FSDP @HF could, that would be great.

Reply:
@AjayP13 the condition `self.is_fsdp_enabled and not _is_peft_model(unwrap_model(model)):` makes sure that inside the else block at the line 2182 we are not having a PEFT model, therefore the condition `if _is_peft_model(unwrap_model(model)):` will never be met, so I think we can remove that whole block. what do you think?

@amyeroberts indeed I think PEFT + FSDP is not well tested, I can take care of adding proper tests on the PEFT testing suite as we do test on transformers main, here: https://github.com/huggingface/peft/blob/main/tests/test_gpu_examples.py we do test PEFT + trainer extensively, I can add a test for FSDP as we do have access to multi-gpu envs on PEFT. Let me know what do you think